.github/workflows/upload-algolia-index.yml

```diff
@@ -0,0 +1,29 @@
+# Uploads index to our Algolia Index
+name: Upload Index
+on:
+  workflow_dispatch:
+  repository_dispatch:
+  schedule:
+    - cron: "0 0 * * *"
```

Comment:
Should we run this in `ibis-docs-main.yml` instead?

Reply:
I guess we could, this doesn't take to long to run.  was keeping it separate because if this fails, it wouldn't messed with the docs build necessarily but, I don't have a strong opinion. I can move it, if you'd prefer so.

Reply:
I think we should move it so that we get CI failures if the script fails instead of it silently failing and us forgetting to look it at for a long time.

We can determine what to do depending on the nature of the failures if/when they happen.